TASVideos/TagHelpers/WikiMarkupTagHelper.cs

```diff
@@ -94,10 +96,22 @@ bool IWriterHelper.CheckCondition(string condition)
 			return null;
 		}
 
+		private static readonly IDictionary<string, Type> ViewComponents = Assembly
+			.GetAssembly(typeof(WikiModuleAttribute))
+			!.GetTypes()
+			.Where(t => t.GetCustomAttribute(typeof(WikiModuleAttribute)) != null)
+			.ToDictionary(tkey => ((WikiModuleAttribute)tkey.GetCustomAttribute(typeof(WikiModuleAttribute))!).Name, tvalue => tvalue);
+
 		void IWriterHelper.RunViewComponent(TextWriter w, string name, string pp)
 		{
 			// TODO: Do we want to asyncify this entire thingy?
-			var content = _viewComponentHelper.InvokeAsync(name, new { pageData = PageData, pp }).Result;
+			var result = ViewComponents.TryGetValue(name, out Type? viewComponent);
+			if (!result)
+			{
+				throw new InvalidOperationException($"Unknown ViewComponent: {name}");

```

Comment:
What happened in the old code without this check?

Reply:
Good question, I'll check it out.  The old code was using the string overload where this one uses the Type overload

Reply:
This isn't hugely important, because theoretically bad modules would have been caught before they got here, but it might be worth digging into.